orchagent/portsorch.cpp

```diff
@@ -6792,3 +6904,37 @@ std::unordered_set<std::string> PortsOrch::generateCounterStats(const string& ty
     }
     return counter_stats;
 }
+
+void PortsOrch::updatePortStateAutoNeg(const Port &port)
+{
+    SWSS_LOG_ENTER();
+
+    if (port.m_type != Port::Type::PHY)
+    {
+        return;
+    }
+
+    string adv_speeds;
+
+    if (getPortAdvSpeeds(port, true, adv_speeds))
+    {
+        m_portStateTable.hset(port.m_alias, "rmt_adv_speeds", adv_speeds);
+    }
+    else
+    {
+        m_port_state_poll[port.m_alias] &= ~PORT_STATE_POLL_AN;
+    }
+}
+
+void PortsOrch::doTask(swss::SelectableTimer &timer)
+{
+    SWSS_LOG_ENTER();
+
+    for (auto it = m_portList.begin(); it != m_portList.end(); ++it)
```

Comment:
why we need loop m_portList? I suppose you can directly loop m_port_state_poll

Reply:
We'll check the port admin state and autoneg config to check if this query should be skipped